### DISCUSSION TITLE
"arg" don't seem to be able to run nested within functions 🤔

### DISCUSSION THREAD

**Discussion by alkene5** [johnlindquist/kit#1374] [Errors] [answered]
```js
// Works
await arg('', ['a', 'b', 'c'])

// Not Works ❌
let func = async () => await arg('', ['a', 'b', 'c'])
func()
```

**Comment by johnlindquist** [comment, marked as answer]
You need to "await func()".

Without await, the script will make it to the "end" of the file and trigger the app the clean it up

**Comment by jdx** [comment]
This should be fixed by #7666.

The issue was that PR #7611 (which fixed the year boundary semver issue) unconditionally sets `target_version_tag`, which bypasses the self_update library's own version comparison. The fix adds an explicit check comparing the current version with the latest version before proceeding with the update.

---
*This comment was AI-generated using Claude Code.*
